src/cycax/cycad/assembly_blender.py

```diff
@@ -0,0 +1,126 @@
+import json
+import logging
+import math
+from pathlib import Path
+
+import bpy
+
+
+class AssemblyBlender:
+    """
+     This class will use the STLs that have been printed and import them to a Blender file that will move them around to their correct location.
+
+    Args:
+        part_no: This is the part number of the complex part that is being assembled.
+
+    """
+
+    def __init__(self, part_no: str) -> None:
+        self.part_no = part_no
+        self._base_path = Path(".")
+        self.parts = []
+
+    def _fetch_part(self, part: str):
+        """
+        Retrieves the part that will be imported and possitioned.
+
+        Args:
+            part: this is the name of the part that will be imported.
+        """
+        stl_file = self._base_path / part / f"{part}.stl"
+        if not stl_file.exists():
+            logging.warning("Referencing a file that does not exists. File name %s", stl_file)
+        bpy.ops.import_mesh.stl(filepath=stl_file)
+        for obj in bpy.context.selected_objects:
+            if part in self.parts:
+                obj.name = f"{part}_{self.parts[part] + 1}"
+                self.parts[part] = self.parts[part] + 1
+            else:
+                obj.name = f"{part}_{1}"
+                self.part[part] = 1
+
+    def _swap_xy_(self, rotation: tuple, rot: float, rotmax: tuple) -> tuple:
+        """Used to help rotate the object on the spot while freezing the top"""
+
+        while rot != 0:
+            max_y = rotmax[1]
+            rotation[0], rotation[1] = max_y - rotation[1], rotation[0]
+            rotmax[0], rotmax[1] = rotmax[1], rotmax[0]
+            rot = rot - 1
+        return rotation, rotmax
+
+    def _swap_xz_(self, rotation: tuple, rot: float, rotmax: tuple) -> tuple:
+        """Used to help rotate the object on the spot while freezing the front"""
+        while rot != 0:
+            max_x = rotmax[0]
+            rotation[0], rotation[2] = rotation[2], max_x - rotation[0]
+            rotmax[0], rotmax[2] = rotmax[2], rotmax[0]
+            rot = rot - 1
+        return rotation, rotmax
+
+    def _swap_yz_(self, rotation: tuple, rot: float, rotmax: tuple) -> tuple:
+        """Used to help rotate the object on the spot while freezing the left"""
+        while rot != 0:
+            max_z = rotmax[2]
+            rotation[1], rotation[2] = max_z - rotation[2], rotation[1]
+            rotmax[2], rotmax[1] = rotmax[1], rotmax[2]
+            rot = rot - 1
+        return rotation, rotmax
+
+    def _move(self, Rotmax: tuple, moves: tuple, Rotate: tuple):
```

Comment:
please use lower case variable names. See the errors mypy reports for this line.

Reply:
SS162